app/scripts/controllers/edit/buildConfig.js

```diff
@@ -154,89 +130,53 @@ angular.module('openshiftConsole')
-                (buildFrom.kind === "ImageStreamTag") ? buildConfig.metadata.namespace + "/" + buildFrom.name : buildFrom.name);
-            } else {
-              $scope.builderOptions = $scope.setPickedVariables($scope.builderOptions, "None", buildConfig.metadata.namespace, "", "", "", "");
-            }
+            var setImageOptions = function(imageOptions, imageData) {
+              imageOptions.type = (imageData && imageData.kind) ? imageData.kind : "None";
+              var ist = {},
+                  isi = "",
+                  di = "";
 
-            if ($scope.updatedBuildConfig.spec.output.to) {
-              var pushTo = $scope.updatedBuildConfig.spec.output.to;
-              $scope.outputOptions = $scope.setPickedVariables(
-                $scope.outputOptions,
-                pushTo.kind,
-                pushTo.namespace || buildConfig.metadata.namespace,
-                pushTo.name.split(":")[0],
-                pushTo.name.split(":")[1],
-                undefined,
-                (pushTo.kind === "ImageStreamTag") ? buildConfig.metadata.namespace + "/" + pushTo.name : pushTo.name);
-            } else {
-              $scope.outputOptions = $scope.setPickedVariables($scope.outputOptions, "None", buildConfig.metadata.namespace, "", "", undefined, "");
-            }
+              if (imageOptions.type === "ImageStreamTag") {
+                ist = {namespace: imageData.namespace || buildConfig.metadata.namespace, imageStream: imageData.name.split(':')[0], tagObject: {tag: imageData.name.split(':')[1]}};
+              } else {
+                ist = {namespace: "", imageStream: "", tagObject: {tag: ""}};
+              }
 
-            $scope.builderImageStream = {
-              namespace: $scope.builderOptions.pickedNamespace,
-              imageStream: $scope.builderOptions.pickedImageStream,
-              tag: $scope.builderOptions.pickedTag,
-            };
+              if (imageOptions.type === "ImageStreamImage") {
+                isi = (imageData.namespace) ? imageData.namespace + "/" + imageData.name : buildConfig.metadata.namespace + "/" + imageData.name;                
+              } else {
+                isi = "";
+              }
+
+              di = imageOptions.type === "DockerImage" ? imageData.name : "";
 
-            $scope.outputImageStream = {
-              namespace: $scope.outputOptions.pickedNamespace,
-              imageStream: $scope.outputOptions.pickedImageStream,
-              tag: $scope.outputOptions.pickedTag,
+              imageOptions.imageStreamTag = ist;
+              imageOptions.imageStreamImage = isi;
+              imageOptions.dockerImage = di;
             };
 
-            $scope.options.forcePull = !!$scope.buildStrategy.forcePull;
+            setImageOptions($scope.imageOptions.from, $scope.buildStrategy.from);
+            setImageOptions($scope.imageOptions.to, $scope.updatedBuildConfig.spec.output.to);
 
             if ($scope.sources.images) {
```

Comment:
Not new, but I'd feel better if we check `_.isEmpty()`. The API omits the field if empty, but I'd prefer to be safe to avoid a runtime error below.